File: frontend/src/types/technician.ts
/**
 * Types for the Technicians admin page. Mirrors
 * ``backend/app/schemas/technician.py``.
 */

export interface Technician {
  id: string;
  name: string;
  phone_e164: string | null;
  whatsapp_chat_jid: string | null;
  is_active: boolean;
  notes: string | null;
  created_at: string;
  updated_at: string | null;
}

export interface TechnicianList {
  items: Technician[];
  total: number;
}

export interface TechnicianCreateInput {
  name: string;
  phone_e164?: string | null;
  whatsapp_chat_jid?: string | null;
  is_active?: boolean;
  notes?: string | null;
}

export interface TechnicianUpdateInput {
  name?: string | null;
  phone_e164?: string | null;
  whatsapp_chat_jid?: string | null;
  is_active?: boolean;
  notes?: string | null;
}

File: backend/app/schemas/technician.py
"""Pydantic schemas for Technician CRUD."""

from uuid import UUID

from pydantic import Field

from app.schemas.base import BaseSchema, TimestampSchema


class TechnicianBase(BaseSchema):
    name: str = Field(..., min_length=1, max_length=255)
    phone_e164: str | None = Field(default=None, max_length=15)
    whatsapp_chat_jid: str | None = Field(default=None, max_length=100)
    is_active: bool = True
    notes: str | None = None


class TechnicianCreate(TechnicianBase):
    """Schema for creating a technician."""


class TechnicianUpdate(BaseSchema):
    """Schema for updating a technician. ``None`` leaves the column unchanged."""

    name: str | None = Field(default=None, min_length=1, max_length=255)
    phone_e164: str | None = Field(default=None, max_length=15)
    whatsapp_chat_jid: str | None = Field(default=None, max_length=100)
    is_active: bool | None = None
    notes: str | None = None


class TechnicianRead(TechnicianBase, TimestampSchema):
    id: UUID


class TechnicianList(BaseSchema):
    items: list[TechnicianRead]
    total: int
